blockchain/chainio.go

```diff
@@ -735,47 +739,37 @@ func dbFetchUtxoEntry(dbTx database.Tx, outpoint wire.OutPoint) (*UtxoEntry, err
 	return entry, nil
 }
 
-// dbPutUtxoView uses an existing database transaction to update the utxo set
-// in the database based on the provided utxo view contents and state.  In
-// particular, only the entries that have been marked as modified are written
-// to the database.
-func dbPutUtxoView(dbTx database.Tx, view *UtxoViewpoint) error {
+// dbPutUtxoEntry uses an existing database transaction to update the utxo entry
+// in the database.
+func dbPutUtxoEntry(dbTx database.Tx, outpoint wire.OutPoint, entry *UtxoEntry) error {
```

Comment:
consider batching this, so that it can write an entire `map[wire.OutPoint]*UtxoEntry`, similar comment for deletion.

in profiling I found that the call to retrieve the utxoset bucket was taking as much time as writing the elements, so batching should amortize this

Reply:
Hmm, so retrieving the bucket is costly? What about just having the bucket in a variable (using sync.Once to make it safe)?

I could batch it, though.